pkg/asset/installconfig/powervs/session.go

```diff
@@ -0,0 +1,251 @@
+package powervs
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/pkg/errors"
+
+	survey "github.com/AlecAivazis/survey/v2"
+	"github.com/IBM-Cloud/power-go-client/ibmpisession"
+
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	//reqAuthEnvs = []string{"IBMID", "IBMID_PASSWORD"}
+	//optAuthEnvs = []string{"IBMCLOUD_REGION", "IBMCLOUD_ZONE"}
+	//debug = false
+	defSessionTimeout time.Duration = 9000000000000000000.0
+	defRegion                       = "us_south"
+)
+
+// Session is an object representing a session for the IBM Power VS API.
+// A bluemix session object may be a better fit here
+type Session struct {
+	Session *ibmpisession.IBMPISession
+}
+
+// PISessionVars is an object that holds the variables required to create an ibmpisession object
+type PISessionVars struct {
+	ID     string `json:"id,omitempty"`
+	APIKey string `json:"apikey,omitempty"`
+	Region string `json:"region,omitempty"`
+	Zone   string `json:"zone,omitempty"`
+}
+
+// GetSession returns an IBM Cloud session by using credentials found in default locations in order:
+// env IBMID & env IBMID_PASSWORD,
+// ~/.bluemix/config.json ? (see TODO below)
+// and, if no creds are found, asks for them
+/* @TODO: if you do an `ibmcloud login` (or in my case ibmcloud login --sso), you get
+//  a very nice creds file at ~/.bluemix/config.json, with an IAMToken. There's no username,
+//  though (just the account's owner id, but that's not the same). It may be necessary
+//  to use the IAMToken vs the password env var mentioned here:
+//  https://github.com/IBM-Cloud/power-go-client#ibm-cloud-sdk-for-power-cloud
+//  Yes, I think we'll need to use the IAMToken. There's a two-factor auth built into the ibmcloud login,
+//  so the password alone isn't enough. The IAMToken is generated as a result. So either:
+     1) require the user has done this already and pull from the file
+     2) ask the user to paste in their IAMToken.
+     3) let the password env var be the IAMToken? (Going with this atm since it's how I started)
+     4) put it into Platform {userid: , iamtoken: , ...}
```

Comment:
can we delete this whole block? this is all pretty outdated brainstorming from months ago :D 
and we should document that users may optionally do `ibmcloud iam api-key-create test --file ~/.bluemix/apikey_test.json --output json` to get that file there.